tests/test_integrated_channels/test_integrated_channel/test_transmitters/test_course_metadata.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+"""
+Tests for the base course metadata transmitter.
+"""
+
+from __future__ import absolute_import, unicode_literals
+
+import unittest
+
+import mock
+from integrated_channels.integrated_channel.transmitters.course_metadata import CourseTransmitter
+from pytest import mark
+
+from test_utils import factories
+
+
+@mark.django_db
+class TestCourseTransmitter(unittest.TestCase):
+    """
+    Tests for the class ``CourseTransmitter``.
+    """
+
+    def setUp(self):
+        super(TestCourseTransmitter, self).setUp()
+        enterprise_customer = factories.EnterpriseCustomerFactory(name='Starfleet Academy')
+        self.enterprise_config = factories.SAPSuccessFactorsEnterpriseCustomerConfigurationFactory(
+            enterprise_customer=enterprise_customer,
+            key="client_id",
+            sapsf_base_url="http://test.successfactors.com/",
+            sapsf_company_id="company_id",
+            sapsf_user_id="user_id",
+            secret="client_secret",
+        )
+        self.global_config = factories.SAPSuccessFactorsGlobalConfigurationFactory()
+
+        # Mocks
+        create_course_content_mock = mock.patch(
+            'integrated_channels.integrated_channel.client.IntegratedChannelApiClient.create_course_content'
+        )
+        self.create_course_content_mock = create_course_content_mock.start()
+        self.create_course_content_mock.return_value = 200, '{}'
+        self.addCleanup(create_course_content_mock.stop)
+        delete_course_content_mock = mock.patch(
+            'integrated_channels.integrated_channel.client.IntegratedChannelApiClient.delete_course_content'
+        )
+        self.delete_course_content_mock = delete_course_content_mock.start()
+        self.delete_course_content_mock.return_value = 200, '{}'
+        self.addCleanup(delete_course_content_mock .stop)
+
+    def test_transmit_block_delete(self):
+        """
+        Passing in 'DELETE' as the method parameter should call the client's ``delete_course_content`` method.
+        """
+        transmitter = CourseTransmitter(self.enterprise_config)
+        transmitter.transmit_block('fake-course-block', method='DELETE')
+        self.delete_course_content_mock.assert_called_once_with('fake-course-block')
```

Comment:
@UmanShahzad Should this module also include a [`test_transmit_block_post`](https://github.com/edx/edx-enterprise/blob/uman/degreed/integrated_channels/integrated_channel/transmitters/course_metadata.py#L95-L96) method? If not, I think you could remove the definition of [`create_course_content_mock`](https://github.com/edx/edx-enterprise/blob/uman/degreed/tests/test_integrated_channels/test_integrated_channel/test_transmitters/test_course_metadata.py#L37-L42) from `setUp` above.

Reply:
:+1: I've used `ddt` to also use `POST`, and the `create_course_content_mock`.

Reply:
@UmanShahzad Nice!